### ISSUE TITLE
pip installation doesn't work

### ISSUE THREAD

**Issue by munechika-koyo** [euratom-software/calcam#103]
The calcam installation with pip command does not successfully done according to the following logs:
<details>
<summary>
❯ pip install -v calcam
</summary>

```bash
Using pip 23.1.2 from /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages/pip (python 3.9)
Collecting calcam
  Using cached calcam-2.10.0.tar.gz (485 kB)
  Running command python setup.py egg_info

  ***************************************************************
                  Calcam v2.10.0 Setup Script
  ***************************************************************

  running egg_info
  creating /tmp/pip-pip-egg-info-7ic6v1je/calcam.egg-info
  writing /tmp/pip-pip-egg-info-7ic6v1je/calcam.egg-info/PKG-INFO
  writing dependency_links to /tmp/pip-pip-egg-info-7ic6v1je/calcam.egg-info/dependency_links.txt
  writing entry points to /tmp/pip-pip-egg-info-7ic6v1je/calcam.egg-info/entry_points.txt
  writing requirements to /tmp/pip-pip-egg-info-7ic6v1je/calcam.egg-info/requires.txt
  writing top-level names to /tmp/pip-pip-egg-info-7ic6v1je/calcam.egg-info/top_level.txt
  writing manifest file '/tmp/pip-pip-egg-info-7ic6v1je/calcam.egg-info/SOURCES.txt'
  reading manifest file '/tmp/pip-pip-egg-info-7ic6v1je/calcam.egg-info/SOURCES.txt'
  adding license file 'LICENCE.txt'
  writing manifest file '/tmp/pip-pip-egg-info-7ic6v1je/calcam.egg-info/SOURCES.txt'

  ***************************************************************

  Preparing metadata (setup.py) ... done
Requirement already satisfied: scipy in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from calcam) (1.11.0)
Requirement already satisfied: matplotlib in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from calcam) (3.5.3)
Requirement already satisfied: cycler>=0.10 in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from matplotlib->calcam) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from matplotlib->calcam) (4.40.0)
Requirement already satisfied: kiwisolver>=1.0.1 in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from matplotlib->calcam) (1.4.4)
Requirement already satisfied: numpy>=1.17 in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from matplotlib->calcam) (1.25.0)
Requirement already satisfied: packaging>=20.0 in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from matplotlib->calcam) (23.1)
Requirement already satisfied: pillow>=6.2.0 in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from matplotlib->calcam) (9.5.0)
Requirement already satisfied: pyparsing>=2.2.1 in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from matplotlib->calcam) (3.1.0)
Requirement already satisfied: python-dateutil>=2.7 in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from matplotlib->calcam) (2.8.2)
Requirement already satisfied: six>=1.5 in /home/koyo/.conda/envs/cherab-phix-dev/lib/python3.9/site-packages (from python-dateutil>=2.7->matplotlib->calcam) (1.16.0)
Building wheels for collected packages: calcam
  Running command python setup.py bdist_wheel
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "/tmp/pip-install-muvhhwto/calcam_0af0b89610a143aeaeb987bfed463aa7/setup.py", line 42, in <module>
      raise Exception("Calcam currently cannot be distributed as wheels due to having important install-time logic in setup.py.")
  Exception: Calcam currently cannot be distributed as wheels due to having important install-time logic in setup.py.
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /home/koyo/.conda/envs/cherab-phix-dev/bin/python3.9 -u -c '
  exec(compile('"'"''"'"''"'"'
  # This is <pip-setuptools-caller> -- a caller that pip uses to run setup.py
  #
  # - It imports setuptools before invoking setup.py, to enable projects that directly
  #   import from `distutils.core` to work with newer packaging standards.
  # - It provides a clear error message when setuptools is not installed.
  # - It sets `sys.argv[0]` to the underlying `setup.py`, when invoking `setup.py` so
  #   setuptools doesn'"'"'t think the script is `-c`. This avoids the following warning:
  #     manifest_maker: standard file '"'"'-c'"'"' not found".
  # - It generates a shim setup.py, for handling setup.cfg-only projects.
  import os, sys, tokenize
  
  try:
      import setuptools
  except ImportError as error:
      print(
          "ERROR: Can not execute `setup.py` since setuptools is not available in "
          "the build environment.",
          file=sys.stderr,
      )
      sys.exit(1)
  
  __file__ = %r
  sys.argv[0] = __file__
  
  if os.path.exists(__file__):
      filename = __file__
      with tokenize.open(__file__) as f:
          setup_py_code = f.read()
  else:
      filename = "<auto-generated setuptools caller>"
      setup_py_code = "from setuptools import setup; setup()"
  
  exec(compile(setup_py_code, filename, "exec"))
  '"'"''"'"''"'"' % ('"'"'/tmp/pip-install-muvhhwto/calcam_0af0b89610a143aeaeb987bfed463aa7/setup.py'"'"',), "<pip-setuptools-caller>", "exec"))' bdist_wheel -d /tmp/pip-wheel-yc56vnav
  cwd: /tmp/pip-install-muvhhwto/calcam_0af0b89610a143aeaeb987bfed463aa7/
  Building wheel for calcam (setup.py) ... error
  ERROR: Failed building wheel for calcam
  Running setup.py clean for calcam
  Running command python setup.py clean

  ***************************************************************
                  Calcam v2.10.0 Setup Script
  ***************************************************************

  running clean
  'build/lib' does not exist -- can't clean it
  'build/bdist.linux-x86_64' does not exist -- can't clean it
  'build/scripts-3.9' does not exist -- can't clean it

  ***************************************************************

Failed to build calcam
ERROR: Could not build wheels for calcam, which is required to install pyproject.toml-based projects
```
</details>

This seems to be caused by the prohibition of `bdist_wheel` in the `setup.py` script.
I guess the latest `pip` forces to build the wheel dists even though using the legacy-installation without `pyproject.toml`.
Anyway we should care this issue for beginning users.

**Comment by munechika-koyo** [comment]
This issue seems to be caused by the deprecation of the fallback source installation by `pip`.
If I changed the `pip` version to `v23.0`, then `calcam` was able to be installed.
Here is the logs:

<details>
<summary>
❯ pip install -v calcam
</summary>

```bash
Using pip 23.0 from /home/koyo/.conda/envs/test/lib/python3.9/site-packages/pip (python 3.9)
Collecting calcam
  Using cached calcam-2.10.0.tar.gz (485 kB)
  Running command python setup.py egg_info

  ***************************************************************
                  Calcam v2.10.0 Setup Script
  ***************************************************************

  running egg_info
  creating /tmp/pip-pip-egg-info-np0ug4q8/calcam.egg-info
  writing /tmp/pip-pip-egg-info-np0ug4q8/calcam.egg-info/PKG-INFO
  writing dependency_links to /tmp/pip-pip-egg-info-np0ug4q8/calcam.egg-info/dependency_links.txt
  writing entry points to /tmp/pip-pip-egg-info-np0ug4q8/calcam.egg-info/entry_points.txt
  writing requirements to /tmp/pip-pip-egg-info-np0ug4q8/calcam.egg-info/requires.txt
  writing top-level names to /tmp/pip-pip-egg-info-np0ug4q8/calcam.egg-info/top_level.txt
  writing manifest file '/tmp/pip-pip-egg-info-np0ug4q8/calcam.egg-info/SOURCES.txt'
  reading manifest file '/tmp/pip-pip-egg-info-np0ug4q8/calcam.egg-info/SOURCES.txt'
  adding license file 'LICENCE.txt'
  writing manifest file '/tmp/pip-pip-egg-info-np0ug4q8/calcam.egg-info/SOURCES.txt'

  ***************************************************************

  Preparing metadata (setup.py) ... done
Requirement already satisfied: scipy in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from calcam) (1.10.1)
Requirement already satisfied: matplotlib in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from calcam) (3.7.1)
Requirement already satisfied: kiwisolver>=1.0.1 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from matplotlib->calcam) (1.4.4)
Requirement already satisfied: python-dateutil>=2.7 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from matplotlib->calcam) (2.8.2)
Requirement already satisfied: pillow>=6.2.0 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from matplotlib->calcam) (9.4.0)
Requirement already satisfied: importlib-resources>=3.2.0 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from matplotlib->calcam) (5.2.0)
Requirement already satisfied: contourpy>=1.0.1 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from matplotlib->calcam) (1.0.5)
Requirement already satisfied: packaging>=20.0 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from matplotlib->calcam) (23.0)
Requirement already satisfied: pyparsing>=2.3.1 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from matplotlib->calcam) (3.0.9)
Requirement already satisfied: cycler>=0.10 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from matplotlib->calcam) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from matplotlib->calcam) (4.25.0)
Requirement already satisfied: numpy>=1.20 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from matplotlib->calcam) (1.25.0)
Requirement already satisfied: zipp>=3.1.0 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from importlib-resources>=3.2.0->matplotlib->calcam) (3.11.0)
Requirement already satisfied: six>=1.5 in /home/koyo/.conda/envs/test/lib/python3.9/site-packages (from python-dateutil>=2.7->matplotlib->calcam) (1.16.0)
Building wheels for collected packages: calcam
  Running command python setup.py bdist_wheel
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "/tmp/pip-install-p67kgg5i/calcam_074e0269596b41719cef593928bce41a/setup.py", line 42, in <module>
      raise Exception("Calcam currently cannot be distributed as wheels due to having important install-time logic in setup.py.")
  Exception: Calcam currently cannot be distributed as wheels due to having important install-time logic in setup.py.
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /home/koyo/.conda/envs/test/bin/python3.9 -u -c '
  exec(compile('"'"''"'"''"'"'
  # This is <pip-setuptools-caller> -- a caller that pip uses to run setup.py
  #
  # - It imports setuptools before invoking setup.py, to enable projects that directly
  #   import from `distutils.core` to work with newer packaging standards.
  # - It provides a clear error message when setuptools is not installed.
  # - It sets `sys.argv[0]` to the underlying `setup.py`, when invoking `setup.py` so
  #   setuptools doesn'"'"'t think the script is `-c`. This avoids the following warning:
  #     manifest_maker: standard file '"'"'-c'"'"' not found".
  # - It generates a shim setup.py, for handling setup.cfg-only projects.
  import os, sys, tokenize
  
  try:
      import setuptools
  except ImportError as error:
      print(
          "ERROR: Can not execute `setup.py` since setuptools is not available in "
          "the build environment.",
          file=sys.stderr,
      )
      sys.exit(1)
  
  __file__ = %r
  sys.argv[0] = __file__
  
  if os.path.exists(__file__):
      filename = __file__
      with tokenize.open(__file__) as f:
          setup_py_code = f.read()
  else:
      filename = "<auto-generated setuptools caller>"
      setup_py_code = "from setuptools import setup; setup()"
  
  exec(compile(setup_py_code, filename, "exec"))
  '"'"''"'"''"'"' % ('"'"'/tmp/pip-install-p67kgg5i/calcam_074e0269596b41719cef593928bce41a/setup.py'"'"',), "<pip-setuptools-caller>", "exec"))' bdist_wheel -d /tmp/pip-wheel-f_cuuh28
  cwd: /tmp/pip-install-p67kgg5i/calcam_074e0269596b41719cef593928bce41a/
  Building wheel for calcam (setup.py) ... error
  ERROR: Failed building wheel for calcam
  Running setup.py clean for calcam
  Running command python setup.py clean

  ***************************************************************
                  Calcam v2.10.0 Setup Script
  ***************************************************************

  running clean
  'build/lib' does not exist -- can't clean it
  'build/bdist.linux-x86_64' does not exist -- can't clean it
  'build/scripts-3.9' does not exist -- can't clean it

  ***************************************************************

Failed to build calcam
Installing collected packages: calcam
  Running command Running setup.py install for calcam

  ***************************************************************
                  Calcam v2.10.0 Setup Script
  ***************************************************************

  Dealing with more difficult dependencies
  before handing over to setuptools...

  OpenCV-Python installation found.
  Compatible PyQt installation found.
  Compatible VTK installation found.

  Now handing over to setuptools...

  running install
  /home/koyo/.conda/envs/test/lib/python3.9/site-packages/setuptools/_distutils/cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
  !!

          ********************************************************************************
          Please avoid running ``setup.py`` directly.
          Instead, use pypa/build, pypa/installer, pypa/build or
          other standards-based tools.

          See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
          ********************************************************************************

  !!
    self.initialize_options()
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/calcam
  copying calcam/misc.py -> build/lib/calcam
  copying calcam/raycast.py -> build/lib/calcam
  copying calcam/pointpairs.py -> build/lib/calcam
  copying calcam/__init__.py -> build/lib/calcam
  copying calcam/cadmodel.py -> build/lib/calcam
  copying calcam/gm.py -> build/lib/calcam
  copying calcam/movement.py -> build/lib/calcam
  copying calcam/io.py -> build/lib/calcam
  copying calcam/image_enhancement.py -> build/lib/calcam
  copying calcam/calibration.py -> build/lib/calcam
  copying calcam/config.py -> build/lib/calcam
  copying calcam/coordtransformer.py -> build/lib/calcam
  copying calcam/render.py -> build/lib/calcam
  creating build/lib/calcam/gui
  copying calcam/gui/image_analysis.py -> build/lib/calcam/gui
  copying calcam/gui/__init__.py -> build/lib/calcam/gui
  copying calcam/gui/vtkinteractorstyles.py -> build/lib/calcam/gui
  copying calcam/gui/core.py -> build/lib/calcam/gui
  copying calcam/gui/cad_edit.py -> build/lib/calcam/gui
  copying calcam/gui/settings.py -> build/lib/calcam/gui
  copying calcam/gui/fitting_calib.py -> build/lib/calcam/gui
  copying calcam/gui/virtual_calib.py -> build/lib/calcam/gui
  copying calcam/gui/qt_wrapper.py -> build/lib/calcam/gui
  copying calcam/gui/alignment_calib.py -> build/lib/calcam/gui
  copying calcam/gui/launch_script.py -> build/lib/calcam/gui
  copying calcam/gui/qvtkrenderwindowinteractor.py -> build/lib/calcam/gui
  copying calcam/gui/movement_correction.py -> build/lib/calcam/gui
  copying calcam/gui/launcher.py -> build/lib/calcam/gui
  copying calcam/gui/viewer.py -> build/lib/calcam/gui
  creating build/lib/calcam/gui/icons
  copying calcam/gui/icons/calcam.png -> build/lib/calcam/gui/icons
  copying calcam/gui/icons/icons8.txt -> build/lib/calcam/gui/icons
  copying calcam/gui/icons/calcam.ico -> build/lib/calcam/gui/icons
  copying calcam/gui/icons/saveas.png -> build/lib/calcam/gui/icons
  copying calcam/gui/icons/info.png -> build/lib/calcam/gui/icons
  copying calcam/gui/icons/open.png -> build/lib/calcam/gui/icons
  copying calcam/gui/icons/save.png -> build/lib/calcam/gui/icons
  copying calcam/gui/icons/calcam-file.ico -> build/lib/calcam/gui/icons
  copying calcam/gui/icons/calcam-file.png -> build/lib/calcam/gui/icons
  copying calcam/gui/icons/new.png -> build/lib/calcam/gui/icons
  creating build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/image_analyser.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/chessboard_image_dialog.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/fitting_calib.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/line_coords.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/rendering_progress.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/movement_corr_dialog.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/split_field_dialog.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/virtual_calib_editor.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/alignment_calib_editor.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/settings.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/launcher.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/cad_edit.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/qt_designer_files/viewer.ui -> build/lib/calcam/gui/qt_designer_files
  copying calcam/gui/logo.png -> build/lib/calcam/gui
  creating build/lib/calcam/builtin_image_sources
  copying calcam/builtin_image_sources/imagefile.py -> build/lib/calcam/builtin_image_sources
  copying calcam/builtin_image_sources/calcam_file.py -> build/lib/calcam/builtin_image_sources
  copying calcam/__version__ -> build/lib/calcam
  copying calcam/gui/__executable_path__ -> build/lib/calcam/gui
  running install_lib
  creating /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/misc.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/raycast.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/__version__ -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/pointpairs.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/__init__.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  creating /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/builtin_image_sources
  copying build/lib/calcam/builtin_image_sources/imagefile.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/builtin_image_sources
  copying build/lib/calcam/builtin_image_sources/calcam_file.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/builtin_image_sources
  copying build/lib/calcam/cadmodel.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/gm.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/movement.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/io.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/image_enhancement.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/calibration.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/config.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  creating /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/image_analysis.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/__init__.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/vtkinteractorstyles.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/core.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/cad_edit.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/settings.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  creating /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/icons/calcam.png -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/icons/icons8.txt -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/icons/calcam.ico -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/icons/saveas.png -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/icons/info.png -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/icons/open.png -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/icons/save.png -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/icons/calcam-file.ico -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/icons/calcam-file.png -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/icons/new.png -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/icons
  copying build/lib/calcam/gui/fitting_calib.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/virtual_calib.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/qt_wrapper.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  creating /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/image_analyser.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/chessboard_image_dialog.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/fitting_calib.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/line_coords.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/rendering_progress.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/movement_corr_dialog.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/split_field_dialog.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/virtual_calib_editor.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/alignment_calib_editor.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/settings.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/launcher.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/cad_edit.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/qt_designer_files/viewer.ui -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_designer_files
  copying build/lib/calcam/gui/alignment_calib.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/logo.png -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/launch_script.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/qvtkrenderwindowinteractor.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/__executable_path__ -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/movement_correction.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/launcher.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/gui/viewer.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui
  copying build/lib/calcam/coordtransformer.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  copying build/lib/calcam/render.py -> /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/misc.py to misc.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/raycast.py to raycast.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/pointpairs.py to pointpairs.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/__init__.py to __init__.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/builtin_image_sources/imagefile.py to imagefile.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/builtin_image_sources/calcam_file.py to calcam_file.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/cadmodel.py to cadmodel.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gm.py to gm.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/movement.py to movement.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/io.py to io.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/image_enhancement.py to image_enhancement.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/calibration.py to calibration.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/config.py to config.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/image_analysis.py to image_analysis.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/__init__.py to __init__.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/vtkinteractorstyles.py to vtkinteractorstyles.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/core.py to core.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/cad_edit.py to cad_edit.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/settings.py to settings.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/fitting_calib.py to fitting_calib.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/virtual_calib.py to virtual_calib.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qt_wrapper.py to qt_wrapper.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/alignment_calib.py to alignment_calib.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/launch_script.py to launch_script.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/qvtkrenderwindowinteractor.py to qvtkrenderwindowinteractor.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/movement_correction.py to movement_correction.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/launcher.py to launcher.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/gui/viewer.py to viewer.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/coordtransformer.py to coordtransformer.cpython-39.pyc
  byte-compiling /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam/render.py to render.cpython-39.pyc
  running install_egg_info
  running egg_info
  writing calcam.egg-info/PKG-INFO
  writing dependency_links to calcam.egg-info/dependency_links.txt
  writing entry points to calcam.egg-info/entry_points.txt
  writing requirements to calcam.egg-info/requires.txt
  writing top-level names to calcam.egg-info/top_level.txt
  reading manifest file 'calcam.egg-info/SOURCES.txt'
  adding license file 'LICENCE.txt'
  writing manifest file 'calcam.egg-info/SOURCES.txt'
  Copying calcam.egg-info to /home/koyo/.conda/envs/test/lib/python3.9/site-packages/calcam-2.10.0-py3.9.egg-info
  running install_scripts
  Installing calcam script to /home/koyo/.conda/envs/test/bin
  writing list of installed files to '/tmp/pip-record-9sy2wwif/install-record.txt'

  ***************************************************************

  Calcam installation complete.

  It can be imported as a Python module with "import calcam"

  The GUI can be launched using the executable:
  /home/koyo/.conda/envs/test/bin/calcam
  or just by typing "calcam" at a terminal.

  ***************************************************************

  Running setup.py install for calcam ... done
  DEPRECATION: calcam was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. pip 23.1 will enforce this behaviour change. A possible replacement is to fix the wheel build issue reported above. Discussion can be found at https://github.com/pypa/pip/issues/8368
Successfully installed calcam-2.10.0
```

</details>

According to the log, legacy-install (i.e. without building wheels?) has deprecated at version 23.1.
Therefore, we should check the pip version before an installation.

**Comment by ssilburn** [comment]
OK, no more hiding from the install method issue then!

There still is no way to have flexible dependencies in the official way of Python packaging, but I think now PyQt5, OpenCV and VTK are available *enough* on different platforms that it's safe enough to abandon setup.py and use the modern way of doing things.

**Comment by munechika-koyo** [comment]
I was looking forward to hearing the good news :+1: 
Regarding the other topics, I would like to recommend that `calcam` should be available for an anaconda user by registering it in a `conda-forge` channel. Unfortunately, I don't really understand how to conduct that, but what do you think about my idea?

**Comment by ssilburn** [comment]
Yeah sounds reasonable to investigate how to also package & release it for conda, I'm not familiar with it either because I don't / haven't used conda so will need to find out how. Please could you open a new separate issue for that? I'll now close this one because the pip compatibility is fixed in release 2.11.
